src/Features/CSharp/Portable/CodeFixes/PopulateSwitch/PopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+            if (!token.Span.IntersectsWith(span))
+            {
+                return null;
+            }
+
+            return (SwitchStatementSyntax)root.FindNode(span);
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var span = context.Span;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var node = GetSwitchStatementNode(root, span);
+            if (node == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    CSharpFeaturesResources.AddSwitchLabels,
+                    (c) => AddMissingSwitchLabelsAsync(model, document, root, node)),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> AddMissingSwitchLabelsAsync(SemanticModel model, Document document, SyntaxNode root, SwitchStatementSyntax switchBlock)
+        {
+            var enumType = (INamedTypeSymbol)model.GetTypeInfo(switchBlock.Expression).Type;
+            var fullyQualifiedEnumType = enumType.ToDisplayString();
+
+            var containsDefaultLabel = false;
+
+            var caseLabels = new List<ExpressionSyntax>();
+            foreach (var section in switchBlock.Sections)
+            {
+                foreach (var label in section.Labels)
+                {
+                    var caseLabel = label as CaseSwitchLabelSyntax;
+                    if (caseLabel != null)
+                    {
+                        caseLabels.Add(caseLabel.Value);
+                        continue;
+                    }
+
+                    if (label.IsKind(SyntaxKind.DefaultSwitchLabel))
+                    {
+                        containsDefaultLabel = true;
+                    }
+                }
+            }
+
+            var missingLabels = GetMissingLabels(model, caseLabels, enumType);
+
+            var breakStatement = SyntaxFactory.BreakStatement();
+            var statements = SyntaxFactory.List(new List<StatementSyntax> { breakStatement });
+
+            var newSections = SyntaxFactory.List(switchBlock.Sections);
+            foreach (var label in missingLabels)
+            {
+                // If an existing simplified label exists, it means we can assume that works already and do it ourselves as well (ergo: there is a static using)
+                var caseLabel =
+                    SyntaxFactory.CaseSwitchLabel(
+                        SyntaxFactory.ParseExpression($"{fullyQualifiedEnumType}.{label}")
+                                     .WithTrailingTrivia(SyntaxFactory.ParseTrailingTrivia(Environment.NewLine)));
```

Comment:
You shoudln't need trailing trivia.  These nodes should have elastic trivia on them by default.  And if they do the formatter should do the right thing.